istio/istio
# Update helm integration tests to use recommended approach when using helm upgrades

Create integration test to use stable revision labels with Helm

Do the following:
1. install helm with 1.10.0 with --set revision=1-10-0 --set
   revisionTags={prod}
2. Install client/server app and communicate between the two of them;
   this will use proxyv2 from 1.10.0
3. install helm with master branch with --set revision=latest --set
   revisionTags={canary}
4. Install client/server app and communicate between the two of them
   (both using latest version)
5. Verify that communication can occur between 1.10.0 and latest with
   echo app
6. Change prod tag to point to the latest revision
7. Change the namespace used for 1-10-0 revision to point to 'latest'
8. Restart the pods in that old namespace. They should now get the
   latest proxyv2 image on restart.
9. Verify that communication between these restarted pods can occur


[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure


Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.
